src/main/NWN/API/Events/Game/DoorEvents.cs

```diff
@@ -60,34 +108,65 @@ public sealed class OnHeartbeat : IEvent
     [GameEvent(EventScriptType.DoorOnLock)]
     public sealed class OnLock : IEvent
     {
+      /// <summary>
+      /// Gets the <see cref="NwDoor"/> that was locked.
+      /// </summary>
       public NwDoor Door { get; } = NWScript.OBJECT_SELF.ToNwObject<NwDoor>();
 
+      /// <summary>
+      /// Gets the LockDC on <see cref="NwPlaceable"/>.
+      /// </summary>
+      public int LockDC { get; } = NWScript.GetLockLockDC(NWScript.OBJECT_SELF);
+
```

Comment:
This can just live in `NwDoor`, as you can specify an object parameter.

I would put `GetLastLocked` in here instead.